### PR TITLE
GDB-8437: introduce a way to expand error messages in GraphDB workbench

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
When exception is thrown after executing of sparql query, the workbench shows the error. But if the error is too big the error area is too long. Most of the users do not need to see the entire message.

## Why
WB displays all error messages no matter of its length.

## How
Added functionality that checks error message length and if it is too long a small part of message is displayed accompanied by a 'Show full error message' button to expand it. When the button is clicked all error will be displayed accompanied by a 'Show less error message' button to show less part of message.

### CODE REVIEW

File: src/js/angular/core/directives/queryeditor/templates/query-editor.html [hunk 1]
```diff
@@ -219,7 +219,24 @@ <h6 class="alert alert-warning no-icon additional" ng-show="!queryIsRunning && !
                     <h6>{{'common.error' | translate}}<span ng-show="yasr.results.getException().status > 0"> {{yasr.results.getException().status}}: {{yasr.results.getException().statusText}}</span></h6>
                 </div>
 
-                <div class="plaintext">{{yasr.results.getException().responseText}}</div>
+                <div ng-if="!fullErrorMessage && sparqlQueryErrorInfo && sparqlQueryErrorInfo.shortenErrorMessage">
+                    <div class="show-full-message">
+                        <a href="#" ng-click="showFullErrorMessage(true)">
+                            <span class="show-full-message-btn">{{'query.editor.error.show.full.message' | translate}}</span>
```
Comment (svilenvelikov) [inline src/js/angular/core/directives/queryeditor/templates/query-editor.html:225]: But it's a link actually.

Comment (svilenvelikov) [reply src/js/angular/core/directives/queryeditor/templates/query-editor.html:225]: Also, why do we need this additional span around the link text?

File: src/js/angular/core/directives/queryeditor/templates/query-editor.html [hunk 2]
```diff
@@ -219,7 +219,24 @@ <h6 class="alert alert-warning no-icon additional" ng-show="!queryIsRunning && !
                     <h6>{{'common.error' | translate}}<span ng-show="yasr.results.getException().status > 0"> {{yasr.results.getException().status}}: {{yasr.results.getException().statusText}}</span></h6>
                 </div>
 
-                <div class="plaintext">{{yasr.results.getException().responseText}}</div>
+                <div ng-if="!fullErrorMessage && sparqlQueryErrorInfo && sparqlQueryErrorInfo.shortenErrorMessage">
+                    <div class="show-full-message">
```
Comment (svilenvelikov) [inline src/js/angular/core/directives/queryeditor/templates/query-editor.html:223]: There are too much containers around these links here. Can we simplify this?

Comment (svilenvelikov) [reply src/js/angular/core/directives/queryeditor/templates/query-editor.html:223]: Also, this `show-full-message` class sounds like something which should be put on some actionable component like button or link, but here it is set on a div tag.

File: src/js/angular/core/directives/queryeditor/templates/query-editor.html (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/core/directives/queryeditor/templates/query-editor.html]: refactored

File: src/js/angular/core/directives/queryeditor/templates/query-editor.html (hunk 2 above)
Comment (boyan-tonchev) [reply src/js/angular/core/directives/queryeditor/templates/query-editor.html]: refactored

**Review by svilenvelikov — APPROVED** [review]

**Review by yordanalexandrov — DISMISSED** [review]

**Review by yordanalexandrov — DISMISSED** [review]

**Review by yordanalexandrov — APPROVED** [review]

**Pull request merged**